### PR TITLE
Revit_Engine side compliance issues

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

This PR resolved Revit_Engine side compliance issues. Engine_Revit_UI is meant to be resolved in a separate PR.

### Test files

Test file on [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FRevit%5FToolkit%2FRevit%5FToolkit%2DIssue573%2DComplianceIssues) showing that the name argument name change gets picked up by the Versioning. All other changes should not require a test file.

### Additional comments

Some descriptions that are touched in this PR are not too informative - they are legacy and need to be revisited each after each, which is beyond the scope of this PR, to be done in the coming weeks/months.

### CODE REVIEW

File: Revit_Engine/Modify/SetFamilyName.cs [hunk 1]
```diff
@@ -33,19 +33,19 @@ public static partial class Modify
...
         /***************************************************/
 
-        [Description("Sets Family Type name for Environment Context Properties")]
+        [Description("Sets Family name for Environment Context Properties")]
         [Input("originContextFragment", "Origin Context Properties")]
-        [Input("familyTypeName", "Revit Family Type Name")]
+        [Input("familyName", "Revit Family Name")]
         [Output("OriginContextFragment")]
-        public static OriginContextFragment SetFamilyTypeName(this OriginContextFragment originContextFragment, string familyTypeName)
+        public static OriginContextFragment SetFamilyName(this OriginContextFragment originContextFragment, string familyName)
```
Comment (kThorsager) [inline Revit_Engine/Modify/SetFamilyName.cs]: This looks script breaking, but it's perhaps negligible?

File: Revit_Engine/Modify/SetFamilyTypeName.cs [hunk 2]
```diff
@@ -38,19 +33,19 @@ public static partial class Modify
...
         /***************************************************/
 
-        [Description("Sets Family name for Environment Context Properties")]
+        [Description("Sets Family Type name for Environment Context Properties")]
         [Input("originContextFragment", "Origin Context Properties")]
-        [Input("familyName", "Revit Family Name")]
+        [Input("familyTypeName", "Revit Family Type Name")]
         [Output("OriginContextFragment")]
-        public static OriginContextFragment SetFamilyName(this OriginContextFragment originContextFragment, string familyName)
+        public static OriginContextFragment SetFamilyTypeName(this OriginContextFragment originContextFragment, string familyTypeName)
```
Comment (kThorsager) [inline Revit_Engine/Modify/SetFamilyTypeName.cs]: Same question as above

File: Revit_Engine/Modify/SetFamilyName.cs (hunk 1 above)
Comment (pawelbaran) [reply Revit_Engine/Modify/SetFamilyName.cs]: I think this is beacuse of how GitHub interpreted the change - actually I swapped file names (`SetFamilyName` was sitting in SetFamilyTypeName.cs_ and `SetFamilyTypeName` in SetFamilyName.cs_ respectively), so the actual methods should not change at all

File: Revit_Engine/Modify/SetFamilyTypeName.cs (hunk 2 above)
Comment (pawelbaran) [reply Revit_Engine/Modify/SetFamilyTypeName.cs]: Same answer as above 😉

**Comment by pawelbaran** [comment]
> and am I miss-remembering that Outputs should start with lowercase?

I have no clue tbh. @FraserGreenroyd are these rules formalized somewhere?

File: Revit_Engine/Modify/SetFamilyName.cs (hunk 1 above)
Comment (kThorsager) [reply Revit_Engine/Modify/SetFamilyName.cs]: Neat, thanks

**Comment by FraserGreenroyd** [comment]
> > and am I miss-remembering that Outputs should start with lowercase?
> 
> I have no clue tbh. @FraserGreenroyd are these rules formalized somewhere?

They are, but current CI checks don't catch them so am happy to leave for this PR if you want? Your choice @pawelbaran though I would prefer lowercase output variable names, it is part of a CI catch for 3.2 at this stage.

**Comment by pawelbaran** [comment]
After a brief look through the code, all output names start with capitals. I am happy to change these, but would rather do that in another PR as the descriptions need revisiting in general (many are missing, some are incomplete, some are legacy). 

I would go for merging this one as long as you do not have a strong position here @FraserGreenroyd, with an assumption I will do my best to resolve #595 ASAP.

**Pull request merged**